### PR TITLE
Restore active preference when reopening window

### LINKED ISSUE #16: Remember the last active tab and show it by default
From the HIG:

> Restore the last viewed preference pane. If the user switches preference panes, your app should remember this change and show the same pane immediately the next time the user opens your preferences window.

We don't currently do that. I think we should do it when users call `.show()` without an argument, and mention in the docs that that should be the preferred way to show the preferences window, and refer to the HIG docs. This would be another benefit of having https://github.com/sindresorhus/Preferences/issues/14 and we could enforce it there.

### PR DESCRIPTION
**Pull request by DivineDominion**
Closes #16

### CODE REVIEW

**Comment by SamusAranX** [comment]
Safari keeps track of the last opened preference pane even after complete app restarts. Based on a cursory glance, I don't think this does.

**Comment by sindresorhus** [comment]
We need to check what the Human Interface Guidelines says about that and what other apps do. I would prefer to not have that behavior, personally.

**Comment by DivineDominion** [comment]
Well:

> **Restore the last viewed preference pane.** If the user switches preference panes, your app should remember this change and show the same pane immediately the next time the user opens your preferences window.
> https://developer.apple.com/design/human-interface-guidelines/macos/app-architecture/preferences/

That could mean either.

[Restoring state](https://developer.apple.com/design/human-interface-guidelines/macos/app-architecture/restoring-state/) is very broad, too, and says "Preserve and restore your app’s state so people can continue where they left off" and "Configure reopened windows as expected".

But what's expected in terms of preferences? I would not expect the app to restore my last preferences editing session, because I don't do that a lot. 

But if an app relies on preferences modifications heavily + requiring restarts, this can get on one's nerves. Does this happen often? At all?

Wouldn't be hard to use UserDefaults to restore the last session, but is that really a use case?

File: Sources/Preferences/PreferencesWindowController.swift [hunk 1]
```diff
@@ -74,15 +74,18 @@ public final class PreferencesWindowController: NSWindowController {
 	///
 	/// - See `close()` to close the window again.
 	/// - See `showWindow(_:)` to show the window without the convenience of activating the app.
-	/// - Note: Unless you need to open a specific pane, prefer not to pass a parameter at all
-	/// - Parameter preferencePane: Identifier of the preference pane to display.
+	/// - Note: Unless you need to open a specific pane, prefer not to pass a parameter at all or `nil`.
+	/// - Parameter preferencePane: Identifier of the preference pane to display, or `nil` to show the
+	///   tab that was open when the user last closed the window.
 	public func show(preferencePane preferenceIdentifier: PreferencePaneIdentifier? = nil) {
```
Comment (sindresorhus) [inline Sources/Preferences/PreferencesWindowController.swift]:
```suggestion
	public func show(preferencePane preferencePaneIdentifier: PreferencePaneIdentifier? = nil) {
```

?

I think we should be consistent in using `preferencePane` over just `preference`.

**Comment by sindresorhus** [comment]
> we start with the interpretation that is least invasive/requires the least amount of code change, i.e. restore the session while the app is running (= this PR)

👍 

> we discuss opening the lib for the other interpretation, e.g. adding an option to store the last open tab in the user defaults, and handle that separately

I don't think we should do this at all. Most of Apple's apps don't do it, neither does most third-party apps. Instead, I think we should open a Radar to get Apple to clarify in the HIG exactly what's expected with restoring. What do you think?

**Comment by DivineDominion** [comment]
Found a bug and fixed it :)  I'm fine with your verdict @sindresorhus as we can still move on merging this PR and then _never_ talk about restoring the tab after relaunching the app. Works for me :)

**Pull request merged**